core/src/test/java/it/pagopa/selfcare/mscore/core/ContractServiceTest.java

```diff
@@ -271,10 +281,124 @@ void testSendDataLakeNotification2() throws ExecutionException, InterruptedExcep
+    }
+
+    /**
+     * Method under test: {@link ContractService#sendDataLakeNotification(Institution, Token, QueueEvent)}
+     */
+    @ParameterizedTest
+    @ValueSource(classes = {
+            MsCoreException.class,
+            ResourceNotFoundException.class
+    })
+    void testSendDataLakeNotification_notOnIpa(Class<?> clazz) throws ExecutionException, InterruptedException {
+        ProducerFactory<String, String> producerFactory = (ProducerFactory<String, String>) mock(ProducerFactory.class);
+        when(producerFactory.transactionCapable()).thenReturn(true);
+        KafkaTemplate<String, String> kafkaTemplate = new KafkaTemplate<>(producerFactory);
+        PagoPaSignatureConfig pagoPaSignatureConfig = new PagoPaSignatureConfig();
+        CoreConfig coreConfig = new CoreConfig();
+        Pkcs7HashSignService pkcs7HashSignService = mock(Pkcs7HashSignService.class);
+        SignatureService signatureService = new SignatureService(new TrustedListsCertificateSource());
+        UserRegistryConnector userRegistryConnector = mock(UserRegistryConnector.class);
+        PartyRegistryProxyConnector partyRegistryProxyConnector = mock(PartyRegistryProxyConnector.class);
+        ContractService contractService = new ContractService(pagoPaSignatureConfig, null, coreConfig,
+                pkcs7HashSignService, signatureService, kafkaTemplate, new KafkaPropertiesConfig(), userRegistryConnector, partyRegistryProxyConnector);
+
+        Institution institution = new Institution();
+        institution.setId("InstitutionId");
+        institution.setExternalId("InstitutionExternalId");
+        institution.setOrigin("IPA");
+        Onboarding onboarding = new Onboarding();
+        onboarding.setProductId("prod");
+        institution.setOnboarding(List.of(onboarding));
+
+        InstitutionUpdate institutionUpdate = new InstitutionUpdate();
+        institutionUpdate.setAddress("42 Main St");
+        institutionUpdate.setBusinessRegisterPlace("Business Register Place");
+        institutionUpdate
+                .setDataProtectionOfficer(new DataProtectionOfficer("42 Main St", "jane.doe@example.org", "Pec"));
+        institutionUpdate.setDescription("The characteristics of someone or something");
+        institutionUpdate.setDigitalAddress("42 Main St");
+        institutionUpdate.setGeographicTaxonomies(new ArrayList<>());
+        institutionUpdate.setImported(true);
+        institutionUpdate.setInstitutionType(InstitutionType.PA);
+        institutionUpdate
+                .setPaymentServiceProvider(new PaymentServiceProvider("Abi Code", "42", "Legal Register Name", "42", true));
+        institutionUpdate.setRea("Rea");
+        institutionUpdate.setShareCapital("Share Capital");
+        institutionUpdate.setSupportEmail("jane.doe@example.org");
+        institutionUpdate.setSupportPhone("6625550144");
+        institutionUpdate.setTaxCode("Tax Code");
+        institutionUpdate.setZipCode("21654");
```

Comment:
avoid replace create for each method, a solution could be creating a simple static method which creates the data object with the default values and reusing this method all over the tests. Look at https://github.com/pagopa/selfcare-ms-core/blob/feature/SELC-2270/core/src/test/java/it/pagopa/selfcare/mscore/core/TestUtils.java